pkg/volume/csi/csi_plugin.go

```diff
@@ -785,13 +798,61 @@ func (p *csiPlugin) skipAttach(driver string) (bool, error) {
 	return false, nil
 }
 
-// getDriverMode returns the driver mode for the specified spec: {persistent|ephemeral}.
+// supportsVolumeMode looks up CSIDriver object associated with the
+// driver name to determine if the driver supports the given volume
+// mode. An error indicates that it isn't supported and explains why.
+func (p *csiPlugin) supportsVolumeMode(driver string, volumeMode csiVolumeMode) error {
```

Comment:
This probably would be better as a `isVolumeSuppoerted(driver, mode) (bool, error)` Returns `true` if supported, `false` if not, and `error` if something went wrong while attempting to check.

Reply:
I had that initially, but then there's no way to convey to the caller what the reason is when the mode isn't supported.

These two error message would have to be removed:
https://github.com/kubernetes/kubernetes/blob/1a14c40a538a6324f867420897d11b4378585941/pkg/volume/csi/csi_plugin.go#L812-L817